torch/distributed/_functional_collectives.py

```diff
@@ -247,8 +293,23 @@ def all_reduce_coalesced(self: List[torch.Tensor], reduceOp: str, group: RANK_TY
     :: N.B. If you pass a PG or a 1D list to perform a MPMD collective, the compiler won't be able to recover
     that information and perform collective algebraic optimization. Use other forms of input for that.
     """
-    tag, rankset, group_size = _expand_group(group, tag)
-    tensor_list = torch.ops.c10d_functional.all_reduce_coalesced(self, reduceOp, tag, rankset, group_size)  # type: ignore[attr-defined]
+    if USE_NATIVE_C10D_FUNCTIONAL:
+        assert isinstance(group, str)
+        assert tag == ""
```

Comment:
why we don't need to call `expand_group_for_native_ops` for allreduce_coalesced?

Reply:
Good catch. I overlooked this